### PR TITLE
Nutanix cloud provider support

### LINKED ISSUE #1131: Implement Nutanix as supported cloud provider
Nutanix is a provider for hyper-converged IT environments on premise. We want to add support for it to machine-controller.

**Acceptance Criteria**
machine-controller can create machines on a Nutanix environment.

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #1131

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Support for Nutanix as cloud provider
```

Signed-off-by: Marvin Beckers <marvin@kubermatic.com>

### CODE REVIEW

File: pkg/cloudprovider/provider/nutanix/provider.go [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+		return nil, errors.New("could not read vm power state")
+	}
+
+	var status instance.Status
+
+	switch *vm.Status.Resources.PowerState {
+	case "ON":
+		status = instance.StatusRunning
+	case "OFF":
+		status = instance.StatusCreating
```
Comment (embik) [inline pkg/cloudprovider/provider/nutanix/provider.go:339]: This is probably wrong but I wasn't sure what status to map to a VM that is off. Any input?

**Comment by embik** [comment]
Test seems to be broken but unrelated.

File: go.mod [hunk 2]
```diff
@@ -33,6 +33,7 @@ require (
 	github.com/prometheus/client_golang v1.11.0
 	github.com/scaleway/scaleway-sdk-go v1.0.0-beta.7
 	github.com/sethvargo/go-password v0.2.0
+	github.com/terraform-providers/terraform-provider-nutanix v1.1.0
```
Comment (kron4eg) [inline go.mod:36]: This module is pointing to the outdated and archived repository, but fresh and updating https://github.com/nutanix/terraform-provider-nutanix has the same module name! which leads to to confusion and problems.

Nutanix don't have dedicated SDK module, but we can extract relevant bits from this terraform provider, using proper mod name and avoid unnecessary dependencies.

TL;DR:
Please don't import terraform provider but instead extract only relevant part (client SDK) into dedicated module.

Comment (embik) [reply go.mod:36]: I agree this isn't pretty and somewhat confusing since the override to the correct location happens later on (https://github.com/kubermatic/machine-controller/pull/1132/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R63). I discussed this with @moadqassem when we started and we determined it was the best thing to do right now.

Nutanix is working on pulling out the code themselves (https://github.com/nutanix/terraform-provider-nutanix/pull/318#issuecomment-991077192), so it didn't seem sensible to do it just to undo it hopefully soon. In addition, the terraform-provider code is licensed under MPL-2.0, I'm not sure we can just copy that code into our codebase.

Comment (kron4eg) [reply go.mod:36]: `github.com/embik/nutanix-client` ;) Once nutanix have their SDK put together, we will switch.

Comment (embik) [reply go.mod]: Done!

**Review by kron4eg — APPROVED** [review]
/approve
